### PR TITLE
feat: Environment Variables from ConfigMap

### PR DESCRIPTION
**Pull request by mattia-fumo**

### CODE REVIEW

File: docs/development_suite/api-console/api-design/services.md [hunk 1]
```diff
@@ -188,6 +188,11 @@ Currently, the following types exist:
...
 * **Plain Text**: values belonging to this type are strings that define the value that we want to associate to that specific environment variable key. For this type it is necessary to specify the **Value** that we want to link to that specific environment variable. **Value** can also be an interpolated string, to do so the value to interpolate needs to be enclosed by two pairs of curly braces, for example `{{VALUE_TO_INTERPOLATE}}`.
 * **From Secret**: this type represents a value that is obtained from a Kubernetes Secret. For this type, it is necessary to specify the **Secret Name** and the **Secret Key** from which this value can be retrieved.
+* **From ConfigMap**: use this type to load a file of a ConfigMap into an environment variable, the **ConfigMap Name** and the **ConfigMap File Name** must be provided.
+
+:::warning
+We always advise mounting ConfigMaps on the files system, and using environment variables *From ConfigMap* only with small-sized files.
+:::
 
 :::info
 If you want more information about adding Kubernetes secrets to a microservice visit the following [link](/development_suite/api-console/api-design/services.md#secrets).
```
Comment (malta895) [inline docs/development_suite/api-console/api-design/services.md:198]: I'd mention also the config map related doc

Comment (mattia-fumo) [reply docs/development_suite/api-console/api-design/services.md]: done

**Review by malta895 — APPROVED** [review]

File: docs/development_suite/api-console/api-design/services.md [hunk 2]
```diff
@@ -188,9 +188,14 @@ Currently, the following types exist:
 
 * **Plain Text**: values belonging to this type are strings that define the value that we want to associate to that specific environment variable key. For this type it is necessary to specify the **Value** that we want to link to that specific environment variable. **Value** can also be an interpolated string, to do so the value to interpolate needs to be enclosed by two pairs of curly braces, for example `{{VALUE_TO_INTERPOLATE}}`.
 * **From Secret**: this type represents a value that is obtained from a Kubernetes Secret. For this type, it is necessary to specify the **Secret Name** and the **Secret Key** from which this value can be retrieved.
+* **From ConfigMap**: use this type to load a file of a ConfigMap into an environment variable, the **ConfigMap Name** and the **ConfigMap File Name** must be provided.
+
+:::warning
+We always advise mounting ConfigMaps on the files system, and using environment variables *From ConfigMap* when necessary and only with small-sized files.
```
Comment (Danielecina) [inline docs/development_suite/api-console/api-design/services.md:194]:
```suggestion
We always suggest using ConfigMaps on the files system and using environment variables *From ConfigMap* when necessary and only with small-sized files.
```

**Review by Danielecina — APPROVED** [review]

File: docs/development_suite/api-console/api-design/services.md [hunk 3]
```diff
@@ -188,9 +188,14 @@ Currently, the following types exist:
 
 * **Plain Text**: values belonging to this type are strings that define the value that we want to associate to that specific environment variable key. For this type it is necessary to specify the **Value** that we want to link to that specific environment variable. **Value** can also be an interpolated string, to do so the value to interpolate needs to be enclosed by two pairs of curly braces, for example `{{VALUE_TO_INTERPOLATE}}`.
 * **From Secret**: this type represents a value that is obtained from a Kubernetes Secret. For this type, it is necessary to specify the **Secret Name** and the **Secret Key** from which this value can be retrieved.
+* **From ConfigMap**: use this type to load a file of a ConfigMap into an environment variable, the **ConfigMap Name** and the **ConfigMap File Name** must be provided.
+
+:::warning
+We always suggest using ConfigMaps on the files system and using environment variables *From ConfigMap* when necessary and only with small-sized files.
```
Comment (mattia-fumo) [inline docs/development_suite/api-console/api-design/services.md:194]:
```suggestion
We always suggest mounting ConfigMaps on the files system and using environment variables *From ConfigMap* when necessary and only with small-sized files.
```

**Pull request merged**